### PR TITLE
[PTQ] Dynamic shapes for OV

### PR DESCRIPTION
**Pull request by KodiaqQ**
### Changes

- Fixed NNCFGraph creation for models with dynamic shapes;
- Fixed post-processing detection pipeline;

### Reason for changes

- Extend the scope of supported models

### Related tickets

- 103935

### CODE REVIEW

File: tests/openvino/native/models.py [hunk 1]
```diff
@@ -225,3 +225,34 @@ def _create_ov_model(self):
         r1 = opset.result(add, name="Result_Add")
         model = ov.Model([r1], [input_1])
         return model
+
+
+@SYNTHETIC_MODELS.register()
```
Comment (l-bat) [inline tests/openvino/native/models.py:230]: If you register `DynamicModel` as `@SYNTHETIC_MODELS`, you must also add `tests/openvino/native/data/reference_scales`.

Comment (KodiaqQ) [reply tests/openvino/native/models.py:230]: Good catch, thanks! Fixed.

**Review by alexsu52 — COMMENTED** [review]
LGTM

**Review by alexsu52 — APPROVED** [review]

**Review by kshpv — APPROVED** [review]
Thanks for the fix!

**Comment by KodiaqQ** [comment]
Jenkins retest this please

**Comment by KodiaqQ** [comment]
run tensorflow pre-commit tests

**Comment by KodiaqQ** [comment]
run pytorch pre-commit tests

**Pull request merged**